Add grafana dashboards for initial build/provision upon component cre…

rcerven opened the pull request:
…ation, simple builds, PaC provisions and unprovisions and triggers

[STONEBLD-1773](https://issues.redhat.com//browse/STONEBLD-1773)
[STONEBLD-1774](https://issues.redhat.com//browse/STONEBLD-1774)

rcerven commented:
/retest

chmeliik commented:
How should we review this? Which parts of the generated config are relevant? Or how could we deploy it and view it ourselves?

rcerven commented:
> How should we review this? Which parts of the generated config are relevant? Or how could we deploy it and view it ourselves?

in diff you can see that it shows 1 file as moved and changed, so I've used "template" from existing dashboards and changed just metric name & those descriptions

chmeliik commented:
> > How should we review this? Which parts of the generated config are relevant? Or how could we deploy it and view it ourselves?
> 
> in diff you can see that it shows 1 file as moved and changed (so there you can see actual changes I've made basically for all of them), so I've used "template" from existing dashboards and changed just metric name & those descriptions

That helps :+1: So all the dashboards are pretty much copy-pastes of each other, just for different metrics?

Still though, if I wanted to deploy it and check it on my cluster, how would I do it? I probably know how to deploy it at least (https://redhat-appstudio.github.io/infra-deployments/docs/development/deployment.html#deploying-your-versions-of-crds) but not sure where to see the dashboard

rcerven commented:
in the cluster find namespace : appstudio-grafana

there check route, and there is url to grafana and there you can see dashboards

tkdchen commented on config/monitoring/grafana-dashboards/grafana-dashboard-simple-build-pipeline.json:
```diff
@@ -0,0 +1,332 @@
+        {
+          "editorMode": "builder",
+          "exemplar": false,
+          "expr": "redhat_appstudio_buildservice_simple_build_pipeline_creation_time_count",
```
So for this example, metric `simple_build_pipeline_creation_time` is exported from build-service controller, and does `redhat_appstudio_buildservice_simple_build_pipeline_creation_time_count` map to that metric?

Does its format follow some Grafana or Promethues expression spec?

Is part `redhat_appstudio_buildservice_` fixed and defined for build-service specifically?

Does part `_count` show the calculation on the collected data of the metric?

rcerven replied:
"Does its format follow some Grafana or Promethues expression spec?"
not sure what you mean, I don't know much about grafana of prometheus

but in the cluster our metric : simple_build_pipeline_creation_time
has to be queried as : redhat_appstudio_buildservice_simple_build_pipeline_creation_time

_count calculates number of metrics
_sum does sum for them
those are prometheus/grafana thingies

tkdchen commented:
/test build-service-e2e

tkdchen commented:
Deployed on my cluster. The dashboards are shown, however after submitting a few builds, both simple and PaC builds, number is still 0 in the dashboard by waiting for a while after the submission. "No data" is shown inside initial build dashboard. Any additional configuration should be done after bootstrapping the cluster?

rcerven commented:
> Deployed on my cluster. The dashboards are shown, however after submitting a few builds, both simple and PaC builds, number is still 0 in the dashboard by waiting for a while after the submission. "No data" is shown inside initial build dashboard. Any additional configuration should be done after bootstrapping the cluster?

I have no idea

I was testing those metrics which we are sending to prometheus in dev cluster and those extended names worked

not sure what has to be done for dashboards

mmorhun reviewed (approved):
LGTM
